pkg/reconciler/ingress/ingress.go

```diff
@@ -196,6 +202,36 @@ func gatherProbes(b *status.Backends, r *gatewayapi.HTTPRoute, visibility v1alph
 	}
 }
 
+// determineLoadBalancerIngressStatus will return the address for the Gateway.
+// If a service is provided, it will return the address of the service.
+// Otherwise, it will return the first address in the Gateway status.
+func (c *Reconciler) determineLoadBalancerIngressStatus(gwc config.GatewayConfig) ([]v1alpha1.LoadBalancerIngressStatus, error) {
+	if gwc.Service != nil {
+		return []v1alpha1.LoadBalancerIngressStatus{
+			{DomainInternal: network.GetServiceHostname(gwc.Service.Name, gwc.Service.Namespace)},
+		}, nil
+	}
+	gw, err := c.gatewayLister.Gateways(gwc.Gateway.Namespace).Get(gwc.Gateway.Name)
+	if apierrs.IsNotFound(err) {
+		return nil, fmt.Errorf("Gateway %s does not exist: %w", gwc.Gateway.Name, err) //nolint:stylecheck
```

Comment:
We should me marking the KIngress - 'Failed' if we can't find the gateway. 

Can you create a follow up issue - don't need to deal with it now.

Reply:
I tried to just handle the marking as failed case.

If you don't think it looks good, I'll revert and make a new issue